### PR TITLE
Added a new component: Stair Carousel

### PR DESCRIPTION
**Pull request by PushpRatan**
I have added a new component stair carousel and here is a little demo of it.

https://github.com/hsnice16/vanilla-web/assets/92618493/c48c1102-ec65-4be1-8f06-eafdec79d3d6

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -67,6 +67,14 @@ <h1 class="hero-text">Vanilla Web</h1>
           Infinite Scrolling Carousel
         </button>
 
+        <!-- Stair carousel -->
```
Comment (hsnice16) [inline index.html]: All the components are arranged in alphabetical order. Please position this component accordingly.

File: stair-carousel/index.html [hunk 2]
```diff
@@ -0,0 +1,93 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Dropdown Animation</title>
```
Comment (hsnice16) [inline stair-carousel/index.html]: Update this title.

File: stair-carousel/script.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  $(prevSecond).removeClass().addClass("prevSecond");
+
+  $(nextSecond).nextAll().removeClass().addClass("hideRight");
+  $(prevSecond).prevAll().removeClass().addClass("hideLeft");
+}
+
+function FindPos(obj) {
+  const posElement = Array.from(obj.parentNode.children).indexOf(obj) + 1;
+  const nbElements = obj.parentNode.childElementCount;
+  console.log(
```
Comment (hsnice16) [inline stair-carousel/script.js]: Do we need this console log here?

File: stair-carousel/script.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  $(next).removeClass().addClass("next");
+
+  $(nextSecond).removeClass().addClass("nextSecond");
+  $(prevSecond).removeClass().addClass("prevSecond");
+
+  $(nextSecond).nextAll().removeClass().addClass("hideRight");
+  $(prevSecond).prevAll().removeClass().addClass("hideLeft");
+}
+
+function FindPos(obj) {
```
Comment (hsnice16) [inline stair-carousel/script.js]: Do we need this function? I think it is only printing a log to the console.
And, if needed, then any reasons to not use the camelCase naming convention here?

File: stair-carousel/styles.css [hunk 5]
```diff
@@ -0,0 +1,156 @@
+/* @charset "utf8";
```
Comment (hsnice16) [inline stair-carousel/styles.css]: If not needed, you can remove this comment.

File: stair-carousel/index.html [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+      <!-- Use of classnames
+      hideleft = img -3 is totally hidden
+      prevSecond = img -2 is reduced and set back
+      prev = img -1 is reduced and set back
+      selected = img is centered and at the first plan
+      next = img +1 is enhanced and brought to the fore
+      nextSecond = img +2 is unhidden and brought to the fore
+      hideRight = img +3 is totally hidden
+    -->
+      <a class="selected" onclick="FindPos(this)" href="javascript:void(0);"
```
Comment (hsnice16) [inline stair-carousel/index.html]: Any reasons to use an anchor tag `<a />` here if we are making them a dead link in the end?

File: stair-carousel/index.html [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+        ><img src="images/inazuma-eleven.webp"
+      /></a>
+      <a class="hideRight" onclick="FindPos(this)" href="javascript:void(0);"
+        ><img src="images/naruto-itachi-uchiha.jpeg"
+      /></a>
+      <a class="hideRight" onclick="FindPos(this)" href="javascript:void(0);"
+        ><img src="images/slam-dunk.jpeg"
+      /></a>
+    </div>
+    <!-- <div class="gradientBRD"></div> -->
```
Comment (hsnice16) [inline stair-carousel/index.html]: If not needed, you can remove this comment.

File: stair-carousel/styles.css [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+
+*, *:before, *:after {
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+}
+
+html, body {
+  margin-top: 10px;
+  /* height: 100%; */
```
Comment (hsnice16) [inline stair-carousel/styles.css]: If not needed, you can remove this comment.

File: stair-carousel/styles.css [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+  border: inherit;
+}
+
+.arrow {
+  cursor: pointer;
+  width: 2.5em;
+  height: 3em;
+}
+
+.left:hover polyline, .left:focus polyline,
```
Comment (hsnice16) [inline stair-carousel/styles.css]: Check if we can use `:is()` function to compact this.

```
:is(.left, .right):is(:hover, :focus) polyline {
...
```

**Review by hsnice16 — CHANGES_REQUESTED** [review]
Great work with the component 🔥. I have just added a few comments.

**Comment by PushpRatan** [comment]
Thanks @hsnice16  for pointing out some of my mistakes. I have changed the code as you suggested.

**Comment by hsnice16** [comment]
Thanks for raising this PR @PushpRatan 🚀

**Pull request merged**